roles/openshift_facts/library/openshift_facts.py

```diff
@@ -914,12 +914,17 @@ def set_sdn_facts_if_unset(facts, system_facts):
 
             if 'networkConfig' in config:
                 if 'clusterNetworkCIDR' in config['networkConfig']:
-                    facts['master']['sdn_cluster_network_cidr'] = \
+                    sdn_cluster_network_cidr = \
                         config['networkConfig']['clusterNetworkCIDR']
                 if 'hostSubnetLength' in config['networkConfig']:
-                    facts['master']['sdn_host_subnet_length'] = \
+                    sdn_host_subnet_length = \
                         config['networkConfig']['hostSubnetLength']
 
+        if 'sdn_cluster_network_cidr' not in facts['master']:
```

Comment:
Any values set by the user will be set before we call `set_sdn_facts_if_unset` and since we have conditionals on the defaults and config values, the config values won't override user values.